src/components/Arguments/index.tsx

```diff
@@ -0,0 +1,254 @@
+  const [ values, setValue ] = useState<IValue>({})
+
+  const errors = validate(list, values)
+  const numberOfErrors = Object.keys(errors).length;
+  const notEnoughSigners = needSigners && selected.length < signers;
+  const haveErrors = numberOfErrors > 0 || notEnoughSigners;
+
+  const [ processingStatus, setProcessingStatus ] = useState(false);
+
+  const [ setResult ] = useSetExecutionResultsMutation();
+  const { scriptFactory, transactionFactory, contractDeployment } =  useTemplateType();
+  const { project, active, isSavingCode } = useProject();
+  const { accounts } = project;
+
+  const signersAccounts = selected.map(i => accounts[i]);
+
+  const send = async ()=>{
+    if (!processingStatus){
+      setProcessingStatus(true);
+    }
+
+    // Map values to strings that will be passed to backend
+    const args = list.map(arg => {
+      const { name, type } = arg
+      return JSON.stringify({ value : values[name], type })
+    })
+
+    let rawResult;
+    try {
+      // Process script
+      if (type === EntityType.ScriptTemplate) {
+        rawResult = await scriptFactory(args);
+      }
+
+      // Process transaction
+      if (type === EntityType.TransactionTemplate){
+        rawResult = await transactionFactory(signersAccounts, args);
+      }
+
+      if (type === EntityType.Account){
+        if (
+            accounts[active.index] &&
+            accounts[active.index].deployedCode
+        ) {
+          if (
+              !confirm("Redeploying will clear the state of all accounts. Proceed?")
+          )
+            setProcessingStatus(false);
+            return;
+        }
+        rawResult = await contractDeployment()
+      }
+
+    } catch (e) {
+      console.error(e)
+      rawResult = e.toString();
+    }
+
+    setProcessingStatus(false);
+    let resultType;
+    switch (type) {
+      case (EntityType.ScriptTemplate):
+        resultType = ResultType.Script
+        break;
+      case (EntityType.TransactionTemplate):
+        resultType = ResultType.Transaction
+        break;
+      default:
+        resultType = ResultType.Contract
+        break;
+    }
+
+    // Display result in the bottom area
+    setResult({
+      variables: {
+        label: project.scriptTemplates[active.index].title,
+        resultType,
+        rawResult
+      }
+    });
+  }
+
+  const isOk = !haveErrors && (validCode !== undefined && !!validCode)
+  let statusIcon = isOk ? <FaRegCheckCircle />: <FaRegTimesCircle/>;
+  let statusMessage = isOk ? "Ready" : "Fix errors";
+
+  const progress = isSavingCode || processingStatus;
+
+  if (progress){
+    statusIcon = <FaSpinner className="spin"/>
+    statusMessage = "Please, wait..."
+  }
+
+  return(
+    <>
+      <div ref={constraintsRef} className="constraints"></div>
+      <motion.div className="drag-box"
+          drag={true}
+          dragConstraints={constraintsRef}
+          dragElastic={1}>
+        <HoverPanel>
+          {list.length > 0 &&
+            <>

```

Comment:
Unnecessary fragment?

Reply:
It's possible to introduce a variable, but I feel it would get complicated in that way.
We want to have a conditional to hide arguments part in case we have a zero arguments in script or transaction.